File: src/main/java/com/example/HealthAndFitnessPlatform/model/Comment.java
package com.example.HealthAndFitnessPlatform.model;


import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "commentTBL")
@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "userId")
    private User user;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "recipeId")
    private Recipe recipe;

    @Lob
    @NotNull(message = "You must write a comment !!")
    private String content;

    @Column(name = "createdAt")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt = LocalDateTime.now();

    @PrePersist
    public void prePersist() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
    }
}

File: HealthAndFitnessPLatformNG/src/app/model/comment.ts
export class Comment {
          
          id:number;
          userId:number;
          recipeId:number;
          content!:string;
          createdAt:string;

          constructor(id:number,userId:number,recipeId:number,content:string,createdAt:string){
                    
                    this.id = id;
                    this.userId = userId;
                    this.recipeId = recipeId;
                    this.content = content;
                    this.createdAt = createdAt;
          }

}